SPARQLWrapper/SmartWrapper.py

```diff
@@ -110,40 +111,37 @@ class Bindings(object):
     :vartype askResult: bool
     """
 
-    def __init__(self, retval):
+    def __init__(self, retval: QueryResult):
         """
         :param retval: the query result.
         :type retval: :class:`QueryResult<SPARQLWrapper.Wrapper.QueryResult>`
         """
         self.fullResult = retval._convertJSON()
         self.head = self.fullResult["head"]
-        self.variables = None
+        self.variables: Optional[List[str]]
         try:
             self.variables = self.fullResult["head"]["vars"]
-        except:
-            pass
+        except KeyError:
+            self.variables = None
 
-        self.bindings = []
-        try:
+        self.bindings: List[Dict[str, Value]] = []
+        if self.variables is not None:
```

Comment:
Looks like this can go into a separate PR.

Reply:
934e334